File: 49. Group Anagrams/groupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans = new ArrayList<>();
        Map<String, List<String>> map = new HashMap<>();

        for (String s : strs) {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String sortedString = new String(chars);

            if (!map.containsKey(sortedString)) {
                map.put(sortedString, new ArrayList<>());
            }

            map.get(sortedString).add(s);
        }

        ans.addAll(map.values());
        return ans;
    }
}

File: 49. Group Anagrams/groupAnagrams.cpp
class Solution
{
public:
    vector<vector<string>> groupAnagrams(vector<string> &strs)
    {

        vector<vector<string>> ans;
        unordered_map<string, vector<string>> m;

        for (auto s : strs)
        {
            string st = s;
            sort(st.begin(), st.end());
            m[st].push_back(s);
        }
        for (auto it : m)
        {
            ans.push_back(it.second);
        }
        return ans;
    }
};